lib/parallel-wait.js

```diff
@@ -0,0 +1,33 @@
+var createCount = require('callback-count');
+
+module.exports = parallelWait;
+
+function parallelWait (/* middlewares */) {
+  var args = Array.prototype.slice.call(arguments);
+
+  return function (req, res, next) {
+    var middlewares = args.slice(); // copy
+    var _err = null;
+    var count = createCount(function(err) { next(_err || err); });
+
+    middlewares.forEach(function () {
+      count.inc(); // inc first just in case the middlewares are sync
```

Comment:
in case of sync middlewares it could be a problem for the same reason for which it is not in the other forEach within the parallel module.
does it make sense?